### PR TITLE
[MIG][15.0] iap_alternative_provider

### PR DESCRIPTION
**Pull request by petrus-v**
- [x] standard migration
- [x] add unit test
- [x] avoid to enforce using `server_environment` to do it in a separate momdule

### CODE REVIEW

**Comment by petrus-v** [comment]
will be used by https://github.com/OCA/connector-telephony/pull/270

**Review by ivantodorovich — APPROVED** [review]

File: iap_alternative_provider/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    "version": "15.0.1.0.0",
+    "category": "Tools",
+    "website": "https://github.com/OCA/server-tools",
+    "author": "Akretion, Odoo Community Association (OCA)",
+    "maintainers": ["sebastienbeau"],
+    "license": "AGPL-3",
+    "application": False,
+    "installable": True,
+    "external_dependencies": {"python": [], "bin": []},
+    "depends": ["iap", "server_environment"],
```
Comment (ivantodorovich) [inline iap_alternative_provider/__manifest__.py]: FTR it might be a good opportunity to split out the `server_environment` dependency in a separate module

**Comment by petrus-v** [comment]
I've moved server_environment concern in a new module in server-env repo https://github.com/OCA/server-env/pull/113

**Review by FernandoRomera — APPROVED** [review]

**Comment by FernandoRomera** [comment]
@ivantodorovich Can you merge this PR, please?

**Comment by AntoniRomera** [comment]
@OCA/community-maintainers 
Can you merge this PR, please?

**Comment by FernandoRomera** [comment]
@nimarosa 
Can you merge this PR, please?

**Comment by FernandoRomera** [comment]
@petrus-v 
Do you know if we can make any progress?

**Comment by nimarosa** [comment]
>  @nimarosa 
> 
> Can you merge this PR, please?
> 
> 

sorry, I can't merge here. I'm not PSC of this repo. Only of human resource ones. I can only merge the modules I maintain in server-tools. 

I think @pedrobaeza have merge access here.

**Comment by pedrobaeza** [comment]
Please rebase and check the error in CI.

**Comment by petrus-v** [comment]
I'll take care of it !

**Comment by petrus-v** [comment]
I think once this PR https://github.com/OCA/server-tools/pull/2503 will be merged It should makes the CI green !

**Comment by nimarosa** [comment]
The letsencrypt error is random. If you rebase probably it will fix CI. Sometimes there are connections errors and letsencrypt module fails. 

Just do another rebase and force push and that will re-run the tests and they should pass.

**Comment by petrus-v** [comment]
> The letsencrypt error is random. If you rebase probably it will fix CI. Sometimes there are connections errors and letsencrypt module fails.
> 
> Just do another rebase and force push and that will re-run the tests and they should pass.

Nice to know, thanks for your input ! in such case mock in unittest should probably improved to avoid such case I'll try to investigate a bit matter adding this on a todo list !

**Pull request merged**